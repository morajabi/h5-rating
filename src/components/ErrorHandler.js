import React, { Component } from 'react'

import { Message } from './Messages'

class ErrorHandler extends Component {
  state = {
    hasError: false,
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: true })
    console.log('Error caught:', error, info)
  }

  render() {
    return this.state.hasError ? (
      <Message>
        Something went wront{' '}
        <span role="img" aria-label="crying face">
          ☹️
        </span>
      </Message>
    ) : (
      this.props.children
    )
  }
}

export default ErrorHandler
