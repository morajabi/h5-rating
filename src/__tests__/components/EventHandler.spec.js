import React from 'react'
import { shallow } from 'enzyme'
import ErrorHandler from '../../components/ErrorHandler'

it('EventHandler renders as expected', () => {
  const wrapper = shallow(<ErrorHandler />)
  expect(wrapper).toMatchSnapshot()
})
