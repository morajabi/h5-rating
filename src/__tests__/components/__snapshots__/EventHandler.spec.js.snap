// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventHandler renders as expected 1`] = `""`;
